packages/SQLServerDataProvider/src/SQLServerDataProvider.ts

```diff
@@ -594,14 +595,14 @@ export class SQLServerDataProvider extends ProviderBase implements IEntityDataPr
         return ProviderType.Database;
     }
 
-    public async GetRecordFavoriteStatus(userId: number, entityName: string, primaryKeyValues: PrimaryKeyValue[]): Promise<boolean> {
-        const id = await this.GetRecordFavoriteID(userId, entityName, primaryKeyValues);
+    public async GetRecordFavoriteStatus(userId: number, entityName: string, compositeKey: CompositeKey): Promise<boolean> {
+        const id = await this.GetRecordFavoriteID(userId, entityName, compositeKey);
         return id !== null;
     }
 
-    public async GetRecordFavoriteID(userId: number, entityName: string, primaryKeyValues: PrimaryKeyValue[]): Promise<number | null> {
+    public async GetRecordFavoriteID(userId: number, entityName: string, compositeKey: CompositeKey): Promise<number | null> {
         try {
-            const sSQL = `SELECT ID FROM [${this.MJCoreSchemaName}].vwUserFavorites WHERE UserID=${userId} AND Entity='${entityName}' AND RecordID='${primaryKeyValues.map(pkv => pkv.Value).join(',')}'`
+            const sSQL = `SELECT ID FROM [${this.MJCoreSchemaName}].vwUserFavorites WHERE UserID=${userId} AND Entity='${entityName}' AND RecordID='${compositeKey.PrimaryKeyValues.map(pkv => pkv.Value).join(',')}'`
```

Comment:
We don't want to do this. What we want to do is build into the `CompositeKey` class the ability for that class to handle in a central place stringify/parse operations so that it can change its representation from object to string and back. Now, here's the deal, right now, we are doing this two ways. Simple comma delim string when we store in the DB and in the case of the URL format in MJ Exp we are doing something else where we use a format:

FieldName|Value||FieldName2|Value2 

And so on. The reason is = has meaning in a URL so we use this single and double vertical pipe. So, to further encasplute that, you move that logic into a method in Composite key called

ToURLSegment() which returns the piped approach and you have ParseURLSegment() which takes in the string and then turns that into the right internal repesentation.

Now, we might be better served by ALWAYS using this piped approach and storing the data in the database that way. Check our code base up and down for all the places where we do the comma delim and perhaps we change them all to do this instead????